File: BOJ/2548.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;
int n, Max;
int arr[20001];
int ans = 999999999;
int ans2;
int idx;
int mid;
int main() {
	scanf("%d", &n);
	for (int i = 1; i <= n; i++)
		scanf("%d", &arr[i]);

	for (int i = 1; i <= n; i++)
		mid += arr[i];
	mid /= n;

	mid += 10;
	for (; mid >= 1;) {
		for (int j = 1; j <= n; j++) {
			if (mid - arr[j] < 0)
				ans2 += (arr[j] - mid);
			else
				ans2 += (mid- arr[j]);
		}
		if (ans >= ans2) {
			ans = ans2;
			idx = mid;
		}
		ans2 = 0;
		mid--;
	}
	printf("%d\n", idx);
	return 0;
}

File: BOJ(Python 공부용)/2548.py
import sys
from collections import deque
from heapq import heappush, heappop
from math import sqrt

input = sys.stdin

if __name__ == '__main__':
    n = int(input.readline())
    v = [int(x) for x in input.readline().split()]
    mid = 0
    for value in v:
        mid += value

    mid //= n
    mid += 10
    idx = 0
    ans, ans2 = int(1e9), 0
    for i in range(mid, 0, -1):
        for value in v:
            ans2 += abs(i - value)

        if ans2 <= ans:
            ans = ans2
            idx = i
        ans2 = 0

    print(idx)
